src/prefect/cli/server.py

```diff
@@ -621,7 +621,10 @@ def start(
         ):
             while not started:
                 try:
-                    client = prefect.Client()
+                    # Get a client with the correct server port
+                    client = prefect.Client(
+                        api_server=f"{config.server.host}:{server_port}"
```

Comment:
I would have liked to reuse `config.server.endpoint` here but because we fill interpolated config values at load-time rather than access-time we cannot set a port in a temporary config :(